recipe_scrapers/_abstract.py

```diff
@@ -65,41 +66,41 @@ def host(cls) -> str:
         """get the host of the url, so we can use the correct scraper"""
         raise NotImplementedError("This should be implemented.")
 
-    def canonical_url(self):
+    def canonical_url(self) -> str | None:
```

Comment:
Is there a reason to prefer `str | None` or `Optional[str]` in some/all cases?

Reply:
oh `str | None` is the new way so `Optional[str]` isn't necessary anymore

Reply:
Ok, great - worth updating the rest of the `Optional[str]` references in this branch to use that too for consistency?

Reply:
Thanks for updating those!